File: backends/fiber/internal/models/ingredient.go
package models

import "github.com/google/uuid"

type Ingredient struct {
	ID            uuid.UUID   `gorm:"column:ingredient_id;primaryKey;default:gen_random_uuid()" json:"id"`
	Name          string      `gorm:"column:name;not null" json:"name"`
	Amount        float64     `gorm:"column:amount;not null" json:"amount"`
	AmountMetric  float64     `gorm:"column:amount_metric;not null" json:"amount_metric"`
	MeasurementID uuid.UUID   `gorm:"column:measurement_id" json:"-"`
	Measurement   Measurement `json:"measurement"`
	RecipeID      uuid.UUID   `gorm:"column:recipe_id" json:"-"`
}

File: backends/aspnet/Recipes.API/entities/Ingredient.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Recipes.API.entities;

public partial class Ingredient
{
    [JsonPropertyName("id")]
    public Guid IngredientId { get; set; }

    public string Name { get; set; } = null!;

    public decimal Amount { get; set; }

    public decimal AmountMetric { get; set; }

    public Guid? MeasurementId { get; set; }

    public Guid? RecipeId { get; set; }

    public virtual Measurement? Measurement { get; set; }

    public virtual Recipe? Recipe { get; set; }

    public virtual ICollection<StepIngredient> StepIngredients { get; set; } = new List<StepIngredient>();
}
